### DISCUSSION TITLE
creating a component from a bunch of vias in HF3D layout

### DISCUSSION THREAD

**Discussion by giuseppeselli** [ansys/pyaedt#3928] [Q&A] [answered]
greetings, 
I have a bunch of via which I am converting into pins by accessing the edb.padstack.padstack_instances[i].in_pin and turning into TRUE. - not sure if there is a better way to do - then I was going to find the name and use Components.create() with the pin names. However, I cannot find the pin name property. Forst of all, I am doing something too cumbersome and there is a better way, if not, where is the pin_name value ?
Thanks,
Giuseppe

**Comment by Samuelopez-ansys** [comment]
@svandenb-dev @ring630

**Comment by svandenb-dev** [comment]
Hi @giuseppeselli,
I created a new example showing how to do this. It will be published once we release a new version. I will notice you when it is available.
Thanks for your feedbacks

**Comment by MaxJPRey** [comment, marked as answer]
See: https://github.com/ansys/pyaedt/pull/3973
